site/helm-charts.md

```diff
@@ -0,0 +1,513 @@
+
+```
+kubernetes/generate-external-weblogic-operator-certificate.sh "DNS:${HOSTNAME},DNS:localhost,IP:127.0.0.1" >> values.yaml
+```
+
+## Optional: ELK (Elasticsearch, Logstash and Kibana) integration
+
+The operator Helm Chart includes the option of installing the necessary Kubernetes resources for ELK integration.
+
+The user is responsible for configuring Kibana and Elasticsearch, then configuring the operator Helm Chart to send events to Elasticsearch. In turn, the operator Helm Chart configures Logstash in the operator deployment to send the operator's log contents to that Elasticsearch location.
+
+### ELK per-operator configuration
+
+As part of the ELK integration, Logstash configuration occurs for each deployed operator instance.  The following properties in values.yaml can be used to configure the integration:
+
+* set "elkIntegrationEnabled" is "true" to enable the integration
+* set "logStashImage" to control which version of logstash is used
+* set "elasticSearchHost" and "elasticSearchPort" to tell the operator where Elasticsearch is running. This will configure Logstash to send the operator's log contents there.
+
+More detailed information about properties in values.yaml can be found in [Operator Helm Chart values.yaml details](#operator-helm-chart-values.yaml-details)
+
+When ELK integration is enabled, the operator's Helm Chart will create a Kubernetes deployment for the operator that will define:
+
+* a Logstash container
+* a persistent volume for the logs
+* a volume mount with mount path "/logs"
+* environment properties for referencing the ElasticSearch resource/pod
+
+## Install the Helm Chart
+
+The "helm install" command is used to install the operator Helm Chart, passing in the user's values.yaml file. As part of this, the user specifies a "release" name for their operator.
+
+The user supplies the "–namespace" argument from the "helm install" command line to specify the namespace in which the operator should be installed.  If not specified, it defaults to "default".  If the namespace does not already exist, Helm will automatically create it (and create a default service account in the new namespace), but will not remove it when the release is deleted.  If the namespace already exists, Helm will re-use it.  These are standard Helm behaviors.
+
+Similarly, the user configures the "serviceAccount" parameter in values.yaml to specify which service account in the operator's namespace that the operator should use.  If not specified, it defaults to "default" (i.e. the namespace's default service account).  If the user wants to use a different service account, then the user must create the operator's namespace and the service account before installing the operator helm chart.
+
+For example:
+
+```
+helm install kubernetes/charts/weblogic-operator --name my-operator --namespace my-operator-namespace --values values.yaml --wait
```

Comment:
it's more typical to call it 'weblogic-operator' (instead of my-operator) since other k8s based products use the term 'operator' too.

Reply:
Done